cpp/ql/test/query-tests/Security/CWE/CWE-190/semmle/TaintedAllocationSize/test.cpp

```diff
@@ -76,7 +76,7 @@ void processData2(char *start, char *end)
 {
 	char *copy;
 
-	copy = new char[end - start]; // GOOD
+	copy = new char[end - start]; // GOOD [FALSE POSITIVE]
```

Comment:
I think this new result is acceptable.  I'm not entirely sure why the old version of the query didn't flag it, and I suspect it was a fluke.

Reply:
It was prevented by the predictability barrier in DefaultTaintTracking, which I did not carry over to this version. The taint source is actually the buffer filled by `fread`, not its return value. I think when #3123 is merged and we can remove pointer/object conflation on arguments, this should be fixed.